slice index out of range panic in zeekio tsv parser

This zeek TSV comes from the zeekio trailing-whitespace.yaml test:
```
#separator \x09
#set_separator	,
#empty_field	(empty)
#unset_field	-
#path	unknown_mime_type_discovery
#open	2020-04-10-18-15-32
#fields	ts	fid	bof
#types	time	string	string
1425634708.454143	FTBt2P3wJBdHGKpZwc
```
Curiously it causes a panic when run wtih `zq -f tzng in.zeek` on master but the unit test on master passes so there is something about the way the ztest module is running this that is avoiding the bug.  I'm disable the test on the dash-z branch and would like to get that merged while this bug can be fixed later...

```
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
github.com/brimsec/zq/zio/zeekio.(*builder).build(0xc0000cf9b0, 0xc000380ea0, 0xc0003595c0, 0x3, 0x4, 0xc00019f170, 0x1b, 0x20, 0xc0003960ae, 0x24, ...)
	/Users/mccanne/repo/zq/zio/zeekio/builder.go:44 +0x8f0
github.com/brimsec/zq/zio/zeekio.(*Parser).ParseValue(0xc0000cf900, 0xc0003960ae, 0x24, 0xff52, 0x0, 0x0, 0x10)
	/Users/mccanne/repo/zq/zio/zeekio/parser.go:329 +0xe5
github.com/brimsec/zq/zio/zeekio.(*Reader).Read(0xc0003796e0, 0x18614c0, 0xc0003796e0, 0x10000)
	/Users/mccanne/repo/zq/zio/zeekio/reader.go:63 +0x205
github.com/brimsec/zq/zio/detector.match(0x1a85540, 0xc0003796e0, 0x191aab2, 0x4, 0x0, 0x0)
	/Users/mccanne/repo/zq/zio/detector/reader.go:104 +0x38
github.com/brimsec/zq/zio/detector.NewReaderWithOpts(0x1a854a0, 0xc000380c60, 0xc0000106e8, 0x7ffeefbffac6, 0x7, 0x191a092, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/mccanne/repo/zq/zio/detector/reader.go:34 +0x595
github.com/brimsec/zq/zio/detector.OpenFromNamedReadCloser(0xc0000106e8, 0x7281fd8, 0xc0000106f0, 0x7ffeefbffac6, 0x7, 0x191a092, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/mccanne/repo/zq/zio/detector/file.go:74 +0x13e
github.com/brimsec/zq/zio/detector.OpenFileWithContext(0x1a95a20, 0xc00003a0e8, 0xc0000106e8, 0x7ffeefbffac6, 0x7, 0x191a092, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/mccanne/repo/zq/zio/detector/file.go:44 +0x218
github.com/brimsec/zq/zio/detector.OpenFile(...)
	/Users/mccanne/repo/zq/zio/detector/file.go:26
github.com/brimsec/zq/cli/inputflags.(*Flags).Open(0xc0002b0da0, 0xc0000106e8, 0xc0000321d0, 0x1, 0x1, 0x1a84a01, 0xc000380b10, 0x0, 0x0, 0x203000, ...)
	/Users/mccanne/repo/zq/cli/inputflags/flags.go:91 +0x198
main.(*Command).Run(0xc0002b0d80, 0xc0000321d0, 0x1, 0x1, 0x0, 0x0)
	/Users/mccanne/repo/zq/cmd/zq/zq.go:159 +0x5b1
github.com/mccanne/charm.(*instance).run(0xc00000ff40, 0xc0000321c0, 0x2, 0x2, 0x0, 0x0)
	/Users/mccanne/go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x293
github.com/mccanne/charm.(*Spec).ExecRoot(0x1ef23e0, 0xc0000321c0, 0x2, 0x2, 0xc00019ff78, 0x1007f65, 0xc000092058, 0x0)
	/Users/mccanne/go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
	/Users/mccanne/repo/zq/cmd/zq/main.go:9 +0x85

```

I just gave it a shot and it repro'ed without a problem. I just cut & pasted the text from the opening description and ran it through the current tip of master, which is `zq` commit f93774d. I've attached that text as [input.log.gz](https://github.com/brimsec/zq/files/6079381/input.log.gz) to save anyone the hazards of their attempted cut & paste missing out on witnessing the repro themselves.

```
$ zq -version
Version: v0.29.0-94-gf93774d0
13:44:18-phil@PhilMac:~/work/zq-2207$ gzcat input.log.gz 
#separator \x09
#set_separator	,
#empty_field	(empty)
#unset_field	-
#path	unknown_mime_type_discovery
#open	2020-04-10-18-15-32
#fields	ts	fid	bof
#types	time	string	string
1425634708.454143	FTBt2P3wJBdHGKpZwc

$ zq -z input.log.gz 
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
github.com/brimsec/zq/zio/zeekio.(*builder).build(0xc000149af0, 0xc000483fb0, 0xc00045b740, 0x3, 0x4, 0xc00019f168, 0x1b, 0x20, 0xc0004b20ae, 0x24, ...)
	/Users/phil/work/zq/zio/zeekio/builder.go:49 +0x999
github.com/brimsec/zq/zio/zeekio.(*Parser).ParseValue(0xc000149a40, 0xc0004b20ae, 0x24, 0xff52, 0x0, 0x0, 0x10)
	/Users/phil/work/zq/zio/zeekio/parser.go:329 +0xe5
github.com/brimsec/zq/zio/zeekio.(*Reader).Read(0xc0004862c0, 0x1870880, 0xc0004862c0, 0x10000)
	/Users/phil/work/zq/zio/zeekio/reader.go:63 +0x205
github.com/brimsec/zq/zio/detector.match(0x1a976e0, 0xc0004862c0, 0x192ab5a, 0x4, 0x0, 0x0)
	/Users/phil/work/zq/zio/detector/reader.go:104 +0x38
github.com/brimsec/zq/zio/detector.NewReaderWithOpts(0x1a961c0, 0xc00014b080, 0xc000010720, 0x7ffeefbffa06, 0xc, 0x192a136, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/phil/work/zq/zio/detector/reader.go:34 +0x595
github.com/brimsec/zq/zio/detector.OpenFromNamedReadCloser(0xc000010720, 0x2f727d8, 0xc000010728, 0x7ffeefbffa06, 0xc, 0x192a136, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/phil/work/zq/zio/detector/file.go:74 +0x13e
github.com/brimsec/zq/zio/detector.OpenFileWithContext(0x1aa7c40, 0xc000036080, 0xc000010720, 0x7ffeefbffa06, 0xc, 0x192a136, 0x4, 0x1, 0x80000, 0xa00000, ...)
	/Users/phil/work/zq/zio/detector/file.go:44 +0x218
github.com/brimsec/zq/zio/detector.OpenFile(...)
	/Users/phil/work/zq/zio/detector/file.go:26
github.com/brimsec/zq/cli/inputflags.(*Flags).Open(0xc0001fda20, 0xc000010720, 0xc0000321d0, 0x1, 0x1, 0x1, 0x0, 0xc000000180, 0xc00019fb28, 0x203000, ...)
	/Users/phil/work/zq/cli/inputflags/flags.go:91 +0x198
main.(*Command).Run(0xc0001fda00, 0xc0000321d0, 0x1, 0x1, 0x0, 0x0)
	/Users/phil/work/zq/cmd/zq/zq.go:158 +0x5f1
github.com/mccanne/charm.(*instance).run(0xc0002ca220, 0xc0000321c0, 0x2, 0x2, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x293
github.com/mccanne/charm.(*Spec).ExecRoot(0x1f0bf60, 0xc0000321c0, 0x2, 0x2, 0xc00019ff78, 0x1007f65, 0xc00010a058, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
	/Users/phil/work/zq/cmd/zq/main.go:9 +0x85
```

In case it wasn't already obvious, I think the explanation is right there in the test YAML. It says:

> The last input line should end with a tab character.

I see now that when I cut & pasted from the issue to my `input.log` the tab did _not_ make the transfer, so, it was invalid Zeek TSV at that point. Indeed, `zeek-cut` says as much:

```
$ gzcat input.log.gz | /usr/local/zeek/bin/zeek-cut 
zeek-cut: skipping log line (not enough fields)
```

So, maybe our to-do is to be equally friendly in rejecting it rather than panicking. 🤷‍♂️

--- a/zio/zeekio/builder.go
+++ b/zio/zeekio/builder.go
@@ -41,8 +41,10 @@ func (b *builder) build(typ *zng.TypeRecord, sourceFields []int, path []byte, da
 		}
 	}
 	b.fields = append(b.fields, data[start:])
-	if len(b.fields) > len(sourceFields) {
+	if actual, expected := len(b.fields), len(sourceFields); actual > expected {
 		return nil, errors.New("too many values")
+	} else if actual < expected {
+		return nil, errors.New("too few values")
 	}
 	b.reorderedFields = b.reorderedFields[:0]
 	for _, s := range sourceFields {

--- a/zio/zeekio/parser_test.go
+++ b/zio/zeekio/parser_test.go
@@ -236,10 +236,15 @@ func TestLegacyZeekInvalid(t *testing.T) {
 	assertError(t, err, "bad types/fields definition", "mismatched #fields/#types headers")
 	*/
 
-	// Test that the wrong number of values is an error
+	// Test that too many values is an error
 	parser = startTest(t, append(standardHeaders, fh, th))
 	_, err = sendLegacyValues(parser, append(values, "extra"))
 	assertError(t, err, "too many values", "wrong number of values")
+
+	// Test that too few values is an error
+	parser = startTest(t, append(standardHeaders, fh, th))
+	_, err = sendLegacyValues(parser, values[:len(values)-2])
+	assertError(t, err, "too few values", "wrong number of values")
 
 	// XXX check invalid types?
 }

